### PR TITLE
Adding checksum to source reference

### PR DESCRIPTION
**Pull request by joedborg**

### CODE REVIEW

**Comment by kyrofa** [comment]
This does indeed seem to be the place to document this feature, but I can't actually find it on https://snapcraft.io/docs/ . @caldav is it missing?

File: reference/plugins/source.md [hunk 1]
```diff
@@ -44,7 +44,12 @@ code for that part, and how to unpack it if necessary.
 
     Snapcraft will checkout the repository or unpack the archive referred to
     by the `source` keyword into `parts/<part-name>/src/` but it will only
-    copy the specified subdirectory into `parts/<part-name>/build/`
+    copy the specified subdirectory into `parts/<part-name>/build/`.
+
+  - `source-checksum`: `<type>/<checksum>`
+
+    Snapcraft will compare the source pulled with the checksum specified
```
Comment (caldav) [inline reference/plugins/source.md]: Suggestion to be at least as complete as the CLI help. 

"Snapcraft will use the checksum specified to verify the integrity of the source. The `source-type` of the source needs to be a file (tar, zip, deb or rpm) and the algorithm either md5, sha1, sha224, sha256, sha384, sha512, sha3_256, sha3_384 or sha3_512. Prepend the checksum with its type, e.g. `sha256/<checksum>`."

**Comment by caldav** [comment]
@kyrofa, looks like it, thanks to both.

**Comment by joedborg** [comment]
@caldav Updated

**Pull request merged**